### PR TITLE
shuttle-cli

### PR DESCRIPTION
**Pull request by jbardin**
Basic cli interface.

### CODE REVIEW

**Comment by jwilder** [comment]
Cool.  I can't tell what the arguments should be for the various options though.  Maybe add a sample invocation or two to the usage docs?

**Comment by jbardin** [comment]
yeah, the help output was really the only time consuming part here. I should at some point refactor it a bit more like how the `go` command works.

**Comment by jbardin** [comment]
maybe this is a little more readable?

File: shuttle-cli/main.go [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+	configFS.StringVar(&cfg.Balance, "balance", "", "balance algorithm, {RR|LC}")
+	configFS.IntVar(&cfg.CheckInterval, "check-interval", 0, "interval between health checks in milliseconds")
+	configFS.IntVar(&cfg.Fall, "fall", 0, "number of failed healthchecks before a backend is marked down")
+	configFS.IntVar(&cfg.Rise, "rise", 0, "number of successful health checks before a down service is marked up")
+	configFS.IntVar(&cfg.ClientTimeout, "client-timeout", 0, "innactivity timeout for client connections")
+	configFS.IntVar(&cfg.ServerTimeout, "server-timeout", 0, "innactivity timeout for server connections")
+	configFS.IntVar(&cfg.DialTimeout, "dial-timeout", 0, "timeout for dialing new connections connections")
+
+	serviceFS.StringVar(&serviceCfg.Addr, "address", "", "service listening address")
+	serviceFS.StringVar(&serviceCfg.Network, "netowrk", "", "service network type")
```
Comment (jwilder) [inline shuttle-cli/main.go]: network

File: shuttle-cli/main.go [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+	serviceFS.IntVar(&serviceCfg.Fall, "fall", 0, "number of failed healthchecks before a backend is marked down")
+	serviceFS.IntVar(&serviceCfg.Rise, "rise", 0, "number of successful health checks before a down service is marked up")
+	serviceFS.IntVar(&serviceCfg.ClientTimeout, "client-timeout", 0, "innactivity timeout for client connections")
+	serviceFS.IntVar(&serviceCfg.ServerTimeout, "server-timeout", 0, "innactivity timeout for server connections")
+	serviceFS.IntVar(&serviceCfg.DialTimeout, "dial-timeout", 0, "timeout for dialing new connections connections")
+	serviceFS.Var(&vhosts, "vhost", "virtual host name. may be set multiple times")
+	serviceFS.Var(&errorPages, "error-page", "location for http error code formatted as 'http://example.com/|500,503'. may be set multiple times")
+
+	backendFS.StringVar(&backendCfg.Addr, "address", "", "service listening address")
+	backendFS.StringVar(&backendCfg.Network, "netowrk", "", "backend network type")
```
Comment (jwilder) [inline shuttle-cli/main.go]: network

Comment (jbardin) [inline shuttle-cli/main.go]: found another too. thx, i'm a little tired :tired_face:

**Comment by jwilder** [comment]
otherwise :+1:

**Comment by jbardin** [comment]
fixed

**Pull request merged**